packages/rulesets/src/asyncapi/functions/asyncApi2ServerVariables.ts

```diff
@@ -0,0 +1,42 @@
+import { createRulesetFunction } from '@stoplight/spectral-core';
+
+import { parseUrlVariables } from './utils/parseUrlVariables';
+import { getMissingProps } from './utils/getMissingProps';
+import { getRedundantProps } from './utils/getRedundantProps';
+
+import type { IFunctionResult } from '@stoplight/spectral-core';
+
+export default createRulesetFunction<{ url: string; variables: Record<string, unknown> }, null>(
+  {
+    input: null,
```

Comment:
```suggestion
    input: {
      type: 'object',
      properties: {
        url: {
          type: 'string',
        },
        variables: {
          type: 'object',
        }
      },
      required: ['url', 'variables'],
    },
```
let's validate the input, so the function is executed only when `url` and `variables` are the values we expect